hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaSingleThreadedMessageReceiver.java

```diff
@@ -106,7 +107,13 @@ public KafkaSingleThreadedMessageReceiver(KafkaConsumer<byte[], byte[]> consumer
                 }
             }
             return Optional.ofNullable(readQueue.poll());
-        } catch (KafkaException ex) {
+        }
+        catch (InterruptException ex ) {
+            logger.debug("Interrupting kafka consumer...", ex);
+            Thread.currentThread().interrupt();
```

Comment:
I think this is one of those rare moments where a comment like
`// KafkaConsumer clears the interrupted status, so we enable it again`

On the other hand, why not rethrow the `ex.getCause()` and handle it in the caller?

Reply:
I just found that constructor of this exception already interrupt the exception, I will study this topic more :) 
'''public class InterruptException extends KafkaException {

    private static final long serialVersionUID = 1L;
    
    public InterruptException(InterruptedException cause) {
        super(cause);
        Thread.currentThread().interrupt();
    }
    
    public InterruptException(String message, InterruptedException cause) {
        super(message, cause);
        Thread.currentThread().interrupt();
    }

    public InterruptException(String message) {
        super(message, new InterruptedException());
        Thread.currentThread().interrupt();
    }

}'''

Reply:
Great find. I'm sincerely surprised by this implementation. Nevertheless it feels like at current stage this invalidates the PR, unless we handle our own exception instead of relying on interrupted state.